### PR TITLE
initial commit of extensions docs

### PR DESCRIPTION
**Pull request by grepory**
A first pass at Extensions documentation.

### CODE REVIEW

File: 0.9/extensions.md [hunk 1]
```diff
@@ -7,4 +7,111 @@ version: 0.9
 
 # Sensu extensions
 
+Extending Sensu is particularly useful in high-volume installations where you
+are running checks, mutating, or handling a large amount of your traffic in a
+particular way.  Rather than forking for each event, extensions run inside the
+main EventMachine loop for Sensu. This should be kept in mind when developing
+extensions for Sensu as you have the "opportunity" to interrupt the event loop
+and cause Sensu and system performance and instability.
```
Comment (portertech) [inline 0.9/extensions.md]: unintentional use of "and"? "performance and instability"

Comment (grepory) [inline 0.9/extensions.md]: Fixed.

Comment (grepory) [inline 0.9/extensions.md]: Oh I misread it again. I think it's just awkward wording. How about:
"to interrupt the event loop and potentially cause instability and degraded performance."

I think it makes more sense if I just leave out "Sensu and system"

File: 0.9/extensions.md [hunk 2]
```diff
@@ -7,4 +7,111 @@ version: 0.9
...
+
+  class Skeleton < Handler # Sub-class the appropriate extension type
+ 
+    # Must at a minimum define type and name. These are used by
+    # Sensu during extension initialization.
+    def definition
+      {
+        type: 'extension',  # Always.
+        name: 'skeleton',   # Usually just class name lower case.
+        mutator: 'amutator' # OPTIONAL
```
Comment (joemiller) [inline 0.9/extensions.md:38]: what does defining a mutator here do?

File: 0.9/extensions.md [hunk 3]
```diff
@@ -7,4 +7,111 @@ version: 0.9
...
+      'Minimal skeleton extension'
+    end
+
+    # run() is passed two hashes: event_data and settings.
+    # for more information, see links below.
+    def run(event_data, settings)
+      # You need to yield to the caller. The first argument should be the
+      # data you want to yield (in the case of handlers, nothing or an error
+      # string, and the return status of the extension.
+      yield("", 0)
```
Comment (joemiller) [inline 0.9/extensions.md:60]: a small blurb or section describing the expected returns for each handler type might be good. I'm assuming mutators yield back a modified event_data, but would be good to be explicit about this

File: 0.9/extensions.md (hunk 2 above)
Comment (joemiller) [inline 0.9/extensions.md:38]: nevermind. i re-read the mutator section at the bottom and got my answer =)

File: 0.9/extensions.md (hunk 3 above)
Comment (joemiller) [inline 0.9/extensions.md:60]: nevermind again. bottom section covers this, it just wasn't immediately obvious. thanks =)

File: 0.9/extensions.md (hunk 2 above)
Comment (grepory) [inline 0.9/extensions.md:38]: In my next revision, I could leave a better comment that addresses both of your questions. The example was meant to be more of a quick reference, so covering those two things would be helpful.

**Pull request merged**